### ISSUE TITLE
add terabox.app

### ISSUE THREAD

**Issue by zYxDevs** [rclone/rclone#7496]
please add terabox.app

thank you🙏

**Comment by Animosity022** [comment]
Unless you plan to implement or sponsor it, very unlikely. Check the backlog list...

**Comment by ytdjtjctyghngtdx** [comment]
Need

**Comment by devnoname120** [comment]
See also: https://github.com/AlistGo/alist/tree/main/drivers/terabox

**Comment by ncw** [comment]
Thanks @b0rrasca that looks like a good API.

**Comment by SirAry** [comment]
Please add Terabox to rclone.
10x

**Comment by Animosity022** [comment]
The way to get it done is to sponsor it, develop or or just wait until a person that wants to maintain the backend picsk it up.

https://rclone.org/sponsor/

There are something like 50+ 'wanted' backends.

**Comment by x1arch** [comment]
coming soon ;)

**Comment by anar4732** [comment]
+++

**Comment by kirmola** [comment]
I mean they kinda use an api, its visible on network interceptor. I made a sample working for me but still, I cant seems to figure out some things, lemme know if you want to see some demo i worked for myself

**Comment by x1arch** [comment]
Sorry guys for delay. The driver almost done, need to be tested complete and write a Documentation. I'll try to make a PR on this weekend.

**Comment by kirmola** [comment]
woooo. lets go

**Comment by jkune5** [comment]
Patiently waiting for this integration

**Comment by computingbee** [comment]
> I mean they kinda use an api, its visible on network interceptor. I made a sample working for me but still, I cant seems to figure out some things, lemme know if you want to see some demo i worked for myself

Their API is documented here, https://www.terabox.com/integrations/docs?lang=en and here is a script that kinda works, https://gist.github.com/nis267/74c6315f6dbd24a0b8889acdd08789e6

**Comment by scascalesp** [comment]
Any news? I can see in oficial doc: https://rclone.org/overview/#supported-backends

**Comment by animosity022** [comment]
> Any news? I can see in oficial doc: https://rclone.org/overview/#supported-backends

The way to get it done is to sponsor it, develop or or just wait until a person that wants to maintain the backend picsk it up.

https://rclone.org/sponsor/

Up to 85 requests for new backends so doesn't seem anyone is interested in picking it up / sponsoring it.

**Comment by x1arch** [comment]
The backend is ready and many users already use it (actually more than 6 month already) you need download developer brunch and build it by yourself while it wouldn't include to main branch.

**Comment by GwynethLlewelyn** [comment]
> The backend is ready and many users already use it (actually more than 6 month already) you need download developer brunch and build it by yourself while it wouldn't include to main branch.

I'm sorry — is your own code (I presume you're talking about your own PR #8508) in any way related to https://github.com/gulp79/rclone-extra ? (I got the link from the rclone forum) It's just that both are very similar (same variable names, same comments, same structure, even same usage of `goto`... eek! lol), but your PR seems to be further developed. 

None of you makes any reference to the other, so I was wondering which one I should use 😁 As said, yours seems to be further developed, so...

@scascalesp if Terabox was displayed [on the supported backends](https://rclone.org/overview/#supported-backends), it's not there any longer; it _might_ be 'soon' (considering that @x1arch's PR only requires a maintainer to approve the final changes (after 130 comments, wow — now that is really thorough, and surely shows how popular/desired the Terabox integration actually is!)

**Comment by x1arch** [comment]
I'm talking about all who really need Terabox already build dev branch and use it. If `rclone-extra` works for you, please use it, I have no any relation with `rclone-extra`. I'm OK if they took my code to their project.

**Comment by gulp79** [comment]
> > The backend is ready and many users already use it (actually more than 6 month already) you need download developer brunch and build it by yourself while it wouldn't include to main branch.
> 
> I'm sorry — is your own code (I presume you're talking about your own PR [#8508](https://github.com/rclone/rclone/pull/8508)) in any way related to https://github.com/gulp79/rclone-extra ? (I got the link from the rclone forum) It's just that both are very similar (same variable names, same comments, same structure, even same usage of `goto`... eek! lol), but your PR seems to be further developed.
> 
> None of you makes any reference to the other, so I was wondering which one I should use 😁 As said, yours seems to be further developed, so...
> 
> [@scascalesp](https://github.com/scascalesp) if Terabox was displayed [on the supported backends](https://rclone.org/overview/#supported-backends), it's not there any longer; it _might_ be 'soon' (considering that [@x1arch](https://github.com/x1arch)'s PR only requires a maintainer to approve the final changes (after 130 comments, wow — now that is really thorough, and surely shows how popular/desired the Terabox integration actually is!)

I used code from Bclone to build my version https://github.com/gulp79/rclone-extra and also to build https://github.com/gulp79/Round-Sync ad i need a rclone with Terabox and Teldrive support.
Now i see that the original code and credits are for [x1arch](https://github.com/x1arch), i'm going to add credits in my readme. 
I've only added plist and little more, otherwise the code (for Terabox) is from x1arch.

**Comment by BenjiThatFoxGuy** [comment]
Hello. I've built bclone as a merge of multiple upstreams to get support for remotes I need. This includes the current state of this PR, where I merge it in whenever a major change is made that looks stable enough to me, that's why I've been following this PR (and the iCloud Photos PR) to pull in some code as it gets pushed in form of an experimental fork of rclone.

I generally merged branches and cherry picked commits in ways that retain the attribution/credit in ""standard git ways""

When I notice rclone-extra, the readme really felt very 1:1 to what I built but I saw a single link to bclone so I guess that's good enough for me. All I did was merge a few upstreams and setup CI (and tested the changes to make sure it all works, with adjustments as necessary). And bclone is MIT anyways just like rclone.

**Comment by gulp79** [comment]
Yes rclone-extra takes the remotes from bclone. I made a fork to better build Round-Sync (Rclone for Android) with Terabox/Teldrive support, as i need libraries for Android.

**Comment by BenjiThatFoxGuy** [comment]
Appreciate your work figuring out the Round Sync btw! I couldn't figure it out myself. 

I'll still update bclone when I have the time, I believe options are nice and I don't plan to obsolete it just yet. Can't wait for terabox to be in the upstream too :3

**Comment by akostadinov** [comment]
which branch exactly supports terabox?

**Comment by gulp79** [comment]
> which branch exactly supports terabox?

https://github.com/x1arch/rclone/tree/backend/terabox  (dev)
or
https://github.com/gulp79/rclone-extra 
or
https://github.com/BenjiThatFoxGuy/bclone
